akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMergeSortedNSpec.scala

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.testkit.Utils._
+import akka.stream.testkit._
+
+import scala.collection.immutable
+import scala.concurrent.duration._
+import akka.stream.testkit.scaladsl.StreamTestKit._
+
+class GraphMergeSortedNSpec extends TwoStreamsSetup {
+  import GraphDSL.Implicits._
+
+  override type Outputs = Int
+
+  override def fixture(b: GraphDSL.Builder[_]): Fixture = new Fixture(b) {
+    val mergeSortedN = b.add(MergeSortedN[Int](2))
+
+    override def left: Inlet[Int] = mergeSortedN.in(0)
+    override def right: Inlet[Int] = mergeSortedN.in(1)
+    override def out: Outlet[Int] = mergeSortedN.out
+  }
+
+  "GraphMergeSortedN" must {
+
+    "work in the happy case" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      RunnableGraph.fromGraph(GraphDSL.create() { implicit b ⇒
+        val mergeSortedN = b.add(MergeSortedN[Int](2))
+
+        Source(1 to 4) ~> mergeSortedN.in(0)
+        Source(2 to 5) ~> mergeSortedN.in(1)
+
+        mergeSortedN.out ~> Sink.fromSubscriber(probe)
+
+        ClosedShape
+      }).run()
+
+      val subscription = probe.expectSubscription()
+
+      subscription.request(4)
+      probe.expectNext(1, 2, 2, 3)
+
+      subscription.request(2)
+      probe.expectNext(3, 4)
+      subscription.request(4)
+      probe.expectNext(4, 5)
+
+      probe.expectComplete()
+    }
+
+    "not complete if one side is available but other completed" in {
```

Comment:
`assertAllStagesStopped` on all of these tests?

Reply:
indeed, this found a bug (there was a case where the stage was not being completed). Thanks